x/stake/endblock.go

```diff
@@ -252,3 +265,63 @@ func handleMatureUnbondingDelegations(k keeper.Keeper, ctx sdk.Context) ([]types
 
 	return completed, events
 }
+
+const (
+	maxProcessedRefundCount = 50
+)
+
+func handleRefundCrossStake(ctx sdk.Context, k keeper.Keeper) sdk.Events {
+	iterator := k.IteratorAllDelegations(ctx)
+	defer iterator.Close()
+	var refundEvents sdk.Events
+	count := 0
+	for ; iterator.Valid(); iterator.Next() {
+		delegation := types.MustUnmarshalDelegation(k.CDC(), iterator.Key(), iterator.Value())
+		if delegation.CrossStake {
+			result := handleMsgSideChainUndelegate(ctx, types.MsgSideChainUndelegate{
```

Comment:
This is not what I mean `send the staking back to BSC`, it wont trigger send all funds back to BSC actually.

Reply:
`handleMsgSideChainUndelegate` will set BeginUnbonding to the store, and the `crossDistributeUndelegated` will be called in `CompleteUnbonding` when doing the endBlock.

<img width="1127" alt="image" src="https://github.com/bnb-chain/bnc-cosmos-sdk/assets/25412254/edbc045e-6fe9-4d9f-b218-b07bf2d9a8f2">

Reply:
`Unbounding` will be trigger in `EndBreatheBlock`